tests/samplers_tests/test_samplers.py

```diff
@@ -84,6 +86,51 @@
         else [lambda: optuna.integration.BoTorchSampler(n_startup_trials=0)]
     ),
 )
+parametrize_reseedable_sampler = copy.deepcopy(parametrize_sampler)
+parametrize_reseedable_sampler.args[1].append(
+    lambda: PartialFixedSampler(
+        fixed_params={"x": 0}, base_sampler=optuna.samplers.RandomSampler()
+    )
+)
```

Comment:
When I added the `PartialFixedSampler` above, a few tests failed as follows:

```
FAILED tests/samplers_tests/test_samplers.py::test_float[distribution3-<lambda>5] - assert False
FAILED tests/samplers_tests/test_samplers.py::test_float[distribution4-<lambda>5] - numpy.core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'subtract' output from dtype('float64') to dtype('int64') with cast...
FAILED tests/samplers_tests/test_samplers.py::test_float[distribution5-<lambda>5] - numpy.core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'subtract' output from dtype('float64') to dtype('int64') with cast...
FAILED tests/samplers_tests/test_samplers.py::test_int[distribution6-<lambda>5] - assert False
FAILED tests/samplers_tests/test_samplers.py::test_categorical[choices0-<lambda>5] - ValueError: '0' not in (1, 2, 3).
FAILED tests/samplers_tests/test_samplers.py::test_categorical[choices1-<lambda>5] - ValueError: '0' not in ('a', 'b', 'c').
FAILED tests/samplers_tests/test_samplers.py::test_categorical[choices2-<lambda>5] - ValueError: '0' not in (1, 'a').
FAILED tests/samplers_tests/test_samplers.py::test_conditional_sample_independent[<lambda>5] - ValueError: '0' not in ('a', 'b').
```

So adding `lambda: PartialFixedSampler(fixed_params={}, base_sampler=optuna.samplers.RandomSampler(),)` to `parametrized` works.

---

I realised that `GridSampler` might also be added to the list. But I suppose that adding `PartialFixedSampler` to the parametrized list is not simple since we need to define`search_space` depending on the tests.

Reply:
> By the way, we have introduced reseed_rng to GridSampler, we can make BaseSampler.reseed_rng as an abstract method. What do you think?

Sounds nice. I'll update this PR to do so.

Reply:
> By the way, we have introduced reseed_rng to GridSampler, we can make BaseSampler.reseed_rng as an abstract method. What do you think?

I think that we need to keep `BaseSampler.reseed_rng` as a non-abstract method because the change requires implementing `reseed_rng` in user-defined samplers.

Reply:
> I think that we need to keep BaseSampler.reseed_rng as a non-abstract method because the change requires implementing reseed_rng in user-defined samplers.

Sounds nice. Let me resolve this conversation.